cocinaborcelle/src/pages/api/cadastrar.ts

```diff
@@ -0,0 +1,55 @@
+import { auth } from "@/../auth/lucia";
+import { Prisma } from "@prisma/client";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+	if (req.method != "POST") {
+		res.status(405);
+		res.end();
+		return;
+	}
+	const { username, password, email } = req.body as {
+		username: string;
+		password: string;
+		email: string;
+	};
+
+	try {
+		const user = await auth.createUser({
+			key: {
+				providerId: "username",
+				providerUserId: email.toLowerCase(),
+				password,
+			},
+			attributes: {
+				nome: username,
+				email: email,
+			},
+		});
+		const session = await auth.createSession({
```

Comment:
```suggestion
		});
		
		await prisma_client.carrinho.create({data:{valor_cents: 0, usuarioId: user.userId}})
		
		const session = await auth.createSession({
```

recomendo criar o carrinho do usuário assim que criar sua conta. Aí, quando ele acessar o seu carrinho, n dará erro e será necessário apenas um show